File: part1-exercises-for-rust/hardways/leetcode-friend-circles/solution.c
void deep(int** M, int MSize, int *visited, int index)
{
    for (int i = 0; i < MSize; i++) {
        if(M[index][i] == 1 && visited[i] == 0) {
            visited[i] = 1;
            deep(M, MSize, visited, i);
        }
    }
}

int findCircleNum(int** M, int MSize, int* MColSize){
    int visited[MSize];
    memset(visited, 0, sizeof(visited));
    int count = 0;
    for (int i = 0; i < MSize; i++) {
        if (visited[i] == 0) {
            deep(M, MSize, visited, i);
            count++;
        }
    }
    return count;
}

File: part1-exercises-for-rust/hardways/leetcode-friend-circles/solution.rs
impl Solution {
    fn deep(m: &Vec<Vec<i32>>,a:&mut Vec<i32>,i:usize){
            for j in 0..m.len(){
                if m[i][j] == 1 && a[j] == 0{
                    a[j] = 1;
                    Solution::deep(m,a,j);
                }
            }
    }

    pub fn find_circle_num(m: Vec<Vec<i32>>) -> i32 {
        let mut a = vec![0;m.len()];
        let mut count = 0;
        for i in 0..m.len(){
            if a[i] ==0 {
                count= count+1;
                Solution::deep(&m,&mut a,i);
            }
        }
        count
    }
}